lib/features/steps/multipart_request_steps.rb

```diff
@@ -0,0 +1,126 @@
+require 'test/unit'
+require 'open-uri'
+require 'json'
+require 'cgi'
+
+include Test::Unit::Assertions
+
+# @!group Multipart request assertion steps
+
+# Verifies that the request contains multipart form-data
+Then('the request contains multipart form-data and corresponding headers') do
+  steps %(
+    Then the "Content-Type" header matches the regex "^multipart\\/form-data; boundary=[\\h-]+$"
+    And the multipart request has a non-empty body
+  )
+end
+
+# Tests the number of fields a multipart request contains.
+#
+# @step_input part_count [Integer] The number of expected fields
+Then('the multipart request has {int} fields') do |part_count|
```

Comment:
I think all of these steps that check `Server.current_request[:body]` aren't specific to multipart requests, so I'm not sure they're needed

My understanding from the source maps tests is that `Server.current_request[:body]` is the parsed request body, so it doesn't matter if it's a JSON or multipart request the parsed body would be the same

So I think these are unnecessary as the existing `the payload field "blah" equals "blah"` steps work for multipart requests already

Reply:
Looking into it a bit further I believe you're correct.  I'll leave the older steps in to ensure backwards compatibility, but remove the newer ones as unnecessary.